GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -0,0 +1,260 @@
+  if (bitmapFontResourceName !== '' && !bitmapFontResource) {
+    console.warn(
+      'Unable to find bitmap font for resource "' +
+        bitmapFontResourceName +
+        '" (or is not a bitmapFont file).'
+    );
+  }
+
+  // There is no bitmap font file or resource related to this resource name.
+  if (!bitmapFontResource || bitmapFontResourceName === '') {
+    // We the default Arial bitmap font
+    return this.getDefaultBitmapFont();
+  }
+
+  // Is case of a resource file is found and is a bitmap font file and this file was preloaded
+  if (
+    this._loadedFontsData[bitmapFontResourceName].fontResourceFileName ===
+    bitmapFontResourceName
+  ) {
+    // Get the fontData of this resource, install the bitmap font and patch the fontName.
+    const fontData = this._loadedFontsData[bitmapFontResourceName].fontData;
+    const bitmapFont = this.patchInstalledBitmapFont(
+      PIXI.BitmapFont.install(fontData, texture)
+    );
+
+    // Finally save the bitmap font previously patched and return it.
+    this._loadedFontsData[bitmapFontResourceName].bitmapFont = bitmapFont;
+    return this._loadedFontsData[bitmapFontResourceName].bitmapFont;
+  }
+};
+
+/**
+ * Check if a bitmap font exist by this name.
+ * The font name is defined in the font file (.fnt/.xml)
+ * @param {string} fontName Name of the bitmap font
+ */
+gdjs.BitmapFontManager.prototype.isBitmapFontLoaded = function (fontName) {
+  return !!PIXI.BitmapFont.available[fontName];
+};
+
+gdjs.BitmapFontManager.prototype.preloadBitmapFontData = async function (
+  onProgress,
+  onComplete
+) {
+  let resources = this._resources;
+
+  const bitmapFontResources = resources.filter(function (resource) {
+    return resource.kind === 'bitmapFont' && !resource.disablePreload;
+  });
+  if (bitmapFontResources.length === 0)
+    return onComplete(bitmapFontResources.length);
+
+  let loaded = 0;
+  /** @type BitmapFontManagerRequestCallback */
+  var onLoad = function (error) {
+    if (error) {
+      console.error('Error while preloading a bitmapFont resource:' + error);
+    }
+
+    loaded++;
+    if (loaded === bitmapFontResources.length) {
+      onComplete(bitmapFontResources.length);
+    } else {
+      onProgress(loaded, bitmapFontResources.length);
+    }
+  };
+
+  for (let i = 0; i < bitmapFontResources.length; ++i) {
+    const response = await fetch(bitmapFontResources[i].file)
+      .then(function (response) {
+        return response;
+      })
+      .catch(function (error) {
+        console.error(
+          "Can't fetch the bitmap font file " +
+            bitmapFontResources[i].file +
+            ', error: ' +
+            error
+        );
+      });
+
+    if (!response) return;
+
+    const fontData = await response.text();
+
+    this._loadedFontsData[bitmapFontResources[i].name] = {
+      fontResourceFileName: bitmapFontResources[i].name,
+      fontData: fontData,
+      bitmapFont: {},
```

Comment:
This should be null. Never use an empty object for anything else than a map. Otherwise, this will just hide logic errors until something blows up later because it's not defined. Instead, set it to `null`, and let the code errors whenever someone tries to use it (it's better for bugs to be visible soon rather than late).
I'm fixing this.